### PR TITLE
feat!: aggregation and resolution picker

### PR DESCRIPTION
**Pull request by corteggiano**
## Overview
add aggregation + resolution dropdowns in the right pane!

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/dashboard/src/components/sidePanel/sections/aggregationSection/helpers.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import type { SelectProps } from '@cloudscape-design/components';
```
Comment (corteggiano) [inline packages/dashboard/src/components/sidePanel/sections/aggregationSection/helpers.tsx:1]: just calling out the files that matter!

File: packages/dashboard/src/components/sidePanel/sections/aggregationSection/index.tsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
+import React from 'react';
```
Comment (corteggiano) [inline packages/dashboard/src/components/sidePanel/sections/aggregationSection/index.tsx:1]: just calling out the files that matter!

File: packages/dashboard/src/hooks/useWidgetDataTypeSet.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import type { SiteWiseAssetQuery } from '@iot-app-kit/source-iotsitewise';
```
Comment (corteggiano) [inline packages/dashboard/src/hooks/useWidgetDataTypeSet.ts:1]: just calling out the files that matter!
hook to get all the data types of the props in a widget

File: packages/dashboard/src/customization/widgets/utils/getAggregationResolution.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { AggregateType } from '@aws-sdk/client-iotsitewise';
```
Comment (corteggiano) [inline packages/dashboard/src/customization/widgets/utils/getAggregationResolution.ts:1]: just calling out the files that matter!
all these defaults are listed in the 5Q doc for aggregations+resolutions

File: packages/dashboard/src/components/sidePanel/sections/aggregationSection/helpers.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+import { AggregateType } from '@aws-sdk/client-iotsitewise';
```
Comment (tracy-french) [inline packages/dashboard/src/components/sidePanel/sections/aggregationSection/helpers.tsx:2]: Nit: Should be type import?

File: packages/dashboard/src/components/sidePanel/sections/aggregationSection/helpers.tsx [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+export const aggregateToString = (aggregate?: string): string => {
+  return aggregate ? aggregate.replace(/_/g, ' ').toLowerCase() : 'auto';
```
Comment (tracy-french) [inline packages/dashboard/src/components/sidePanel/sections/aggregationSection/helpers.tsx:5]: It's not immediately obvious why there is the call to `replace`. Do you think a comment would help?

File: packages/dashboard/src/components/sidePanel/sections/aggregationSection/index.tsx [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+import type { QueryWidget } from '~/customization/widgets/types';
+import { AssetPropertyQuery, AssetQuery } from '@iot-app-kit/source-iotsitewise/dist/es/time-series-data/types';
+import type { DashboardWidget } from '~/types';
+import { getResolutionOptions, getAggregationOptions } from './helpers';
+import { useWidgetDataTypeSet } from '~/hooks/useWidgetDataTypeSet';
+
+import './index.css';
+
+export const isAggregationSupported = (widget: DashboardWidget): widget is QueryWidget =>
+  ['line-chart', 'scatter-chart', 'bar-chart', 'status-timeline', 'table', 'kpi', 'status'].some(
```
Comment (tracy-french) [inline packages/dashboard/src/components/sidePanel/sections/aggregationSection/index.tsx:18]: Nit: This may read better if this list is abstracted into a constant.

File: packages/dashboard/src/components/sidePanel/sections/aggregationSection/index.tsx [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+  );
+
+const defaultMessages = {
+  header: 'Aggregations and resolution',
+  resolutionLabel: 'Resolution',
+  aggregationMessage: 'Aggregation',
+};
+
+const getAggregationSectionOptions = (siteWiseAssetQuery: SiteWiseAssetQuery | undefined, widget: QueryWidget) => {
+  const { aggregationType, resolution } = siteWiseAssetQuery?.assets[0]?.properties[0] || {};
```
Comment (tracy-french) [inline packages/dashboard/src/components/sidePanel/sections/aggregationSection/index.tsx:29]: Nit: Prefer `??` over `||`.

File: packages/dashboard/src/hooks/useWidgetDataTypeSet.ts [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+    .map(({ assetId, properties }) =>
+      properties.map(
+        ({ propertyId }) =>
+          describedAssetsMap[assetId]?.assetProperties
+            ?.filter((prop) => prop.id === propertyId)
+            .map((prop) => prop.dataType) || []
+      )
+    )
+    .flat(2)
+    .filter(isDefined);
```
Comment (tracy-french) [inline packages/dashboard/src/hooks/useWidgetDataTypeSet.ts:18]: This is hard to follow due to the high density of logic. Consider improving the readability.

**Review by tracy-french — CHANGES_REQUESTED** [review]
Overall, your work looks great @corteggiano! Just a few minor comments, mainly pointing out areas where readability may be improved. I would also like to see some integration tests to show everything working together.

Thank you for building this feature!

File: packages/dashboard/src/components/sidePanel/sections/aggregationSection/helpers.tsx (hunk 5 above)
Comment (corteggiano) [reply packages/dashboard/src/components/sidePanel/sections/aggregationSection/helpers.tsx:2]: its actually the enum that we're importing in
image

File: packages/dashboard/src/customization/widgets/kpi/component.tsx [hunk 10]
```diff
@@ -30,6 +32,7 @@ const KPIWidgetComponent: React.FC<KPIWidget> = (widget) => {
   const { iotSiteWiseQuery } = useQueries();
   const query = iotSiteWiseQuery && queryConfig.query ? iotSiteWiseQuery?.timeSeriesData(queryConfig.query) : undefined;
   const key = computeQueryConfigKey(viewport, queryConfig);
+  const aggregation = getAggregation(queryConfig);
```
Comment (tracy-french) [inline packages/dashboard/src/customization/widgets/kpi/component.tsx:35]: Clean!

File: packages/dashboard/src/customization/widgets/utils/widgetAggregationUtils.ts [hunk 11]
```diff
@@ -0,0 +1,37 @@
+import { AggregateType } from '@aws-sdk/client-iotsitewise';
+import { AssetQuery } from '@iot-app-kit/source-iotsitewise/dist/es/time-series-data/types';
+import { SiteWiseQueryConfig } from '../types';
+
+export const WidgetDefaultAggregation: Record<string, AggregateType | undefined> = {
```
Comment (tracy-french) [inline packages/dashboard/src/customization/widgets/utils/widgetAggregationUtils.ts:5]: Nice type annotation!

**Review by tracy-french — APPROVED** [review]
Excellent, high-quality production code. SOLID, clean, and documented new feature for IoT App Kit. Thank you for your contribution @corteggiano!

**Review by jmbuss — APPROVED** [review]
